lib/ContextModule.js

```diff
@@ -12,7 +12,7 @@ const DepBlockHelpers = require("./dependencies/DepBlockHelpers");
 const Template = require("./Template");
 
 class ContextModule extends Module {
-	constructor(resolveDependencies, context, recursive, regExp, addon, asyncMode, chunkName) {
+	constructor(resolveDependencies, context, recursive, regExp, addon, asyncMode, chunkName, contextDependency) {
```

Comment:
You can't pass the contextDependency through the ContextModule. All options affecting the content of the ContextModule must be visible in the `identifier()`.

* Split `contextDependency` into `filter` and `exlude`.

Now have a lot of arguments in this constructor and as this is a breaking change anyway, why not creating a options object.

* Change constructor to `constructor(resolveDependencies, options)`
* Rename `context` to `resource` to align it with the signature of `resolveDependencies`
* Add the new options to `identifier()` `readableIdentifier()` and `libIdent()`